projects/core/src/occ/services/occ-fields.service.ts

```diff
@@ -0,0 +1,179 @@
+import { Injectable } from '@angular/core';
+import { extractFields, mergeFields, parseFields } from '../utils/occ-fields';
+import { ScopedData } from '../../model/scoped-data';
+import { map, shareReplay } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+
+/**
+ * Intermediate model to accommodate all data needed to perform occ fields optimizations
+ * wrapping ScopedData with url and fields
+ */
+export interface OccFieldsModels {
+  /** Url (with fields) to load scoped data */
+  url?: string;
+  /** extracted fields object, used to extract data from broader model */
+  fields?: object;
+  /** scoped data model */
+  model: ScopedData<any>;
+}
+
+/**
+ * Grouped rest calls with optimal urls
+ *
+ * One url groups all scopes it covers with related occFieldsModels
+ */
+export interface OccMergedUrls {
+  [optimalUrl: string]: {
+    [scope: string]: OccFieldsModels;
+  };
+}
+
+/**
+ * Helper service for optimizing endpoint calls to occ backend
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class OccFieldsService {
+  constructor(protected http: HttpClient) {}
+
+  protected FIELDS_PARAM = 'fields';
+
+  /**
+   * Optimize occ endpoint calls merging requests to the same url by merging field parameters
```

Comment:
Can we comment here what will be returned when we pass as arguments various products (with different product codes) and various scopes? How it will be grouped and optimized?

Reply:
ok